code/iff_defs/iff_defs.cpp

```diff
@@ -300,24 +287,17 @@ void iff_init()
 		}
 
 		// begin reading data
-		Num_iffs = 0;
 		while (required_string_either("#End", "$IFF Name:"))
 		{
 			iff_info *iff;
 			int cur_iff;
 
-			// make sure we're under the limit
-			if (Num_iffs >= MAX_IFFS)
-			{
-				Warning(LOCATION, "Too many iffs in iffs_defs.tbl!  Max is %d.\n", MAX_IFFS);
-				skip_to_start_of_string("#End", NULL);
-				break;
-			}
-
 			// add new IFF
-			iff = &Iff_info[Num_iffs];
-			cur_iff = Num_iffs;
-			Num_iffs++;
+			cur_iff = (int) Iff_info.size();
+			Iff_info.push_back(iff_info());
+			attack_names.push_back(decltype(attack_names)::value_type());
+			observed_color_table.push_back(decltype(observed_color_table)::value_type());
```

Comment:
The `decltype(attack_names)::value_type()` looks weird but maybe it's just because I don't see it often.  But what about using a simple `emplace_back()`?